tests/lib/ocpp/v201/test_database_handler.cpp

```diff
@@ -65,4 +65,91 @@ TEST_F(DatabaseHandlerV201, KO1_FR27_DatabaseWithProfileData_InsertNewProfile) {
     ASSERT_EQ(count, 2);
 }
 
+TEST_F(DatabaseHandlerV201, KO1_FR27_DatabaseWithProfileData_DeleteRemovesSpecifiedProfiles) {
+    db_handler->insert_or_update_charging_profile(1, ChargingProfile{.id = 1, .stackLevel = 1});
+    db_handler->insert_or_update_charging_profile(1, ChargingProfile{.id = 2, .stackLevel = 1});
+
+    auto sql = "SELECT COUNT(*) FROM CHARGING_PROFILES";
+
+    auto select_stmt = db_connection->new_statement(sql);
+
+    do {
+        ASSERT_TRUE(select_stmt->step() == SQLITE_ROW);
+        auto count = select_stmt->column_int(0);
+        ASSERT_EQ(count, 2);
+    } while (select_stmt->step() != SQLITE_DONE);
+
+    db_handler->delete_charging_profile(1);
+
+    do {
+        ASSERT_TRUE(select_stmt->step() == SQLITE_ROW);
+        auto count = select_stmt->column_int(0);
+        ASSERT_EQ(count, 1);
+    } while (select_stmt->step() != SQLITE_DONE);
+}
+
+TEST_F(DatabaseHandlerV201, KO1_FR27_DatabaseWithProfileData_DeleteAllRemovesAllProfiles) {
+    db_handler->insert_or_update_charging_profile(1, ChargingProfile{.id = 1, .stackLevel = 1});
+    db_handler->insert_or_update_charging_profile(1, ChargingProfile{.id = 2, .stackLevel = 1});
+
+    auto sql = "SELECT COUNT(*) FROM CHARGING_PROFILES";
+
+    auto select_stmt = db_connection->new_statement(sql);
+
+    do {
+        ASSERT_TRUE(select_stmt->step() == SQLITE_ROW);
+        auto count = select_stmt->column_int(0);
+        ASSERT_EQ(count, 2);
+    } while (select_stmt->step() != SQLITE_DONE);
+
+    db_handler->delete_charging_profiles();
+
+    do {
+        ASSERT_TRUE(select_stmt->step() == SQLITE_ROW);
+        auto count = select_stmt->column_int(0);
+        ASSERT_EQ(count, 0);
+    } while (select_stmt->step() != SQLITE_DONE);
+}
+
+TEST_F(DatabaseHandlerV201, KO1_FR27_DatabaseWithNoProfileData_DeleteAllDoesNotFail) {
+
+    auto sql = "SELECT COUNT(*) FROM CHARGING_PROFILES";
+
+    auto select_stmt = db_connection->new_statement(sql);
+
+    do {
```

Comment:
Perhaps refactor this out into a function called ensure_db_has_num_profiles(int count)?

Reply:
This is an issue that is present because of the wrapper around Sqlite. This adds a little visibility to the issue. Since this is most likely only needed in test I did not refactor.